orttraining/orttraining/python/experimental/orttrainer_options.py

```diff
@@ -95,6 +95,22 @@ class ORTTrainerOptions(object):
                         }
                     }
                 },
+                'graph_transformer': {
```

Comment:
We should be very deliberate about what goes in to the API. Otherwise we run the risk of bloating the API a lot. Some questions that may help set context to make a decision:
1) Why is this needed? What use cases does this unblock?
2) How is a user expected to take a decision on this? 
3) Is there a way to set this automatically?
4) Is there a similar argument or precedent for this in PyTorch?